tests/ord-aggregator/tests/registry_client.go

```diff
@@ -0,0 +1,159 @@
+package tests
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/accessstrategy"
+	httputil "github.com/kyma-incubator/compass/components/director/pkg/http"
+	"github.com/kyma-incubator/compass/components/director/pkg/log"
+	"github.com/pkg/errors"
+	"github.com/tidwall/gjson"
+)
+
+type Client interface {
+	GetGlobalProductsAndVendorsNumber(ctx context.Context, globalRegistryURL string) (int, int, error)
+}
+
+type client struct {
+	*http.Client
+	accessStrategyExecutorProvider accessstrategy.ExecutorProvider
+}
+
+func NewClient(httpClient *http.Client, accessStrategyExecutorProvider accessstrategy.ExecutorProvider) *client {
+	return &client{
+		Client:                         httpClient,
+		accessStrategyExecutorProvider: accessStrategyExecutorProvider,
+	}
+}
+
+func (c *client) GetGlobalProductsAndVendorsNumber(ctx context.Context, globalRegistryURL string) (int, int, error) {
+	config, err := c.fetchConfig(ctx, globalRegistryURL)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	baseURL, err := calculateBaseURL(globalRegistryURL, *config)
+	if err != nil {
+		return 0, 0, errors.Wrap(err, "while calculating baseURL")
+	}
+
+	globalProducts := 0
+	globalVendors := 0
+	for _, docDetails := range config.OpenResourceDiscoveryV1.Documents {
+		documentURL, err := buildDocumentURL(docDetails.URL, baseURL)
+		if err != nil {
+			return globalProducts, globalVendors, errors.Wrap(err, "error building document URL")
+		}
+		strategy, ok := docDetails.AccessStrategies.GetSupported()
+		if !ok {
+			log.C(ctx).Warnf("Unsupported access strategies for ORD Document %q", documentURL)
+			continue
+		}
```

Comment:
Better return error here as the tests need this. In the aggregator, we continue because in productive use-cases some of the documents that we can fetch are still more valuable than no documents. However, in the test scenario, you need all documents, otherwise, the test will fail later with a difference in expected numbers which will be harder to debug.